src/cmakeProject.ts

```diff
@@ -1541,20 +1541,21 @@ export class CMakeProject {
             }
         } else {
             // single file with known path
-            const compdbPath = path.join(await this.binaryDir, 'compile_commands.json');
+            const compdbPath = util.lightNormalizePath(path.join(await this.binaryDir, 'compile_commands.json'));
             if (await fs.exists(compdbPath)) {
                 compdbPaths.push(compdbPath);
                 if (this.workspaceContext.config.copyCompileCommands) {
                     // Now try to copy the compdb to the user-requested path
-                    const copyDest = this.workspaceContext.config.copyCompileCommands;
-                    const expandedDest = await expandString(copyDest, opts);
+                    const copyDest = util.lightNormalizePath(this.workspaceContext.config.copyCompileCommands);
+                    const expandedDest = util.lightNormalizePath(await expandString(copyDest, opts));
                     if (compdbPath !== expandedDest) {
                         const parentDir = path.dirname(expandedDest);
                         try {
                             log.debug(localize('copy.compile.commands', 'Copying {2} from {0} to {1}', compdbPath, expandedDest, 'compile_commands.json'));
                             await fs.mkdir_p(parentDir);
                             try {
                                 await fs.copyFile(compdbPath, expandedDest);
+                                compdbPaths.push(expandedDest);
```

Comment:
curious why we're also adding this path? It should be the same file as the previously added file in line 1546?

Reply:
No, it's not the same file. The one on line 1546 is where CMake / the build system outputs the compile_commands.json to start with. However, the `expandedDest` is wherever the user has defined in the setting `cmake.copyCompileCommands`, which could be somewhere different.